File: ts/src/test/Exchange/test.fetchMarginModes.ts
import assert from 'assert';
import { Exchange } from "../../../ccxt";
import testMarginMode from './base/test.marginMode.js';
import testSharedMethods from './base/test.sharedMethods.js';

async function testFetchMarginModes (exchange: Exchange, skippedProperties: string[], symbol: string) {
    const method = 'fetchMarginModes';
    const marginModes = await exchange.fetchMarginModes ([ 'symbol' ]);
    assert (typeof marginModes === 'object', exchange.id + ' ' + method + ' ' + symbol + ' must return an object. ' + exchange.json (marginModes));
    const marginModeKeys = Object.keys (marginModes);
    testSharedMethods.assertNonEmtpyArray (exchange, skippedProperties, method, marginModes, symbol);
    for (let i = 0; i < marginModeKeys.length; i++) {
        const marginMode = marginModes[marginModeKeys[i]];
        testSharedMethods.assertNonEmtpyArray (exchange, skippedProperties, method, marginMode, symbol);
        testMarginMode (exchange, skippedProperties, method, marginMode);
    }
}

export default testFetchMarginModes;

File: julia/Ccxt/test/exchange/test.fetchMarginModes.jl
using Test
using Ccxt
function testFetchMarginModes(exchange, skippedProperties, symbol)

    method = "fetchMarginModes";
    marginModes = Base.fetch(fetchMarginModes(exchange, ["symbol"]));
    @test functions.ccxtruthy(isDictionary(exchange, marginModes))
    marginModeKeys = objectKeys(marginModes);
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, marginModes, symbol);
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(marginModeKeys)))
        marginMode = get(marginModes, Symbol(get(marginModeKeys, i + 1, nothing)), nothing);
        assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, marginMode, symbol);
        testMarginMode(exchange, skippedProperties, method, marginMode);
        i += 1
    end
    return true
end
